### PR TITLE
Discover turbo policy through controller runtime client

### PR DESCRIPTION
**Pull request by ading1977**
This PR implements the following:
* Update the go module and go vendor to reflect the dependency on controller runtime
* Create a runtime scheme, register and add the custom types with the scheme
* Create a controller runtime client that supports the custom types
* Implement the `TurbolicyProcessor` that can get the turbo **policies** and turbo **policybindings** 

Note:
* This PR will be updated before merging to reflect the `turbo-crd` changes that are still in review
* Converting the discovered policies and policybindings into Turbonomic group and policy will be covered in a separate PR

Commit [8fc7acd](https://github.com/turbonomic/kubeturbo/pull/694/commits/8fc7acda71b4cff30c6ccb0c9d4d0eaae80b2ad9) has the most relevant changes.

### CODE REVIEW

File: pkg/discovery/processor/cluster_processor.go [hunk 1]
```diff
@@ -182,11 +182,15 @@ func (p *ClusterProcessor) DiscoverCluster() (*repository.ClusterSummary, error)
...
 
-	// Discover volumes
+	// Discover Volumes
 	NewVolumeProcessor(p.clusterInfoScraper, kubeCluster).ProcessVolumes()
 
+	// Discover Business Apps
 	NewBusinessAppProcessor(p.clusterInfoScraper, kubeCluster).ProcessBusinessApps()
 
+	// Discover Turbo Policies
+	NewTurbolicyProcessor(p.clusterInfoScraper, kubeCluster).ProcessTurboPolicies()
```
Comment (pankajbatra1) [inline pkg/discovery/processor/cluster_processor.go:192]: Typo: NewTurboPolicyProcessor

File: pkg/discovery/processor/turbo_policy_processor.go [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+	sloScalings, err := p.ClusterScraper.GetAllTurboSLOScalings()
+	if err != nil {
+		glog.Warningf("Failed to list SLOHorizontalScales: %v.", err)
+		return
+	}
+	if len(sloScalings) == 0 {
+		glog.V(2).Info("There is no SLOHorizontalScale resource found in the cluster.")
+		return
+	}
+	glog.V(3).Infof("Turbo SLOHorizontalScales: %v.", spewConfigState.Sdump(sloScalings))
```
Comment (irfanurrehman) [inline pkg/discovery/processor/turbo_policy_processor.go:42]: What to do with these discovered policies is being kept out of scope of this PR, right?

File: pkg/discovery/processor/cluster_processor.go (hunk 1 above)
Comment (ading1977) [reply pkg/discovery/processor/cluster_processor.go]: Thanks for catching this!

File: pkg/discovery/processor/turbo_policy_processor.go (hunk 2 above)
Comment (ading1977) [reply pkg/discovery/processor/turbo_policy_processor.go:42]: Yes, converting the discovered policies into Turbo group and policy will be handled in a separate PR.
Integration test will also be added in a separate PR.

File: pkg/cluster/cluster_info_scraper.go [hunk 3]
```diff
@@ -50,21 +55,25 @@ type ClusterScraperInterface interface {
...
 type ClusterScraper struct {
 	*client.Clientset
-	caClient      *capiclient.Clientset
-	capiNamespace string
-	DynamicClient dynamic.Interface
-	cache         turbostore.ITurboCache
+	caClient                *capiclient.Clientset
+	capiNamespace           string
+	DynamicClient           dynamic.Interface
+	ControllerRuntimeClient runtimeclient.Client
```
[the remark this answers is not in the document]
Comment (ading1977) [reply pkg/cluster/cluster_info_scraper.go:67]: I feel the benefit of doing this right now is not big. I think we can handle this together when we replace dynamic client with controller runtime client.

**Review by pankajbatra1 — APPROVED** [review]

**Pull request merged**